Few updates for your cool ansible role

dnikolayev opened the pull request:
Hello, David

I am not a big fan of glassfish but was playing with it using ansible. This pull request has small fixes and updates.

1) Found strange problem with permissions to files:
TASK [dpalomar.glassfish : install Glassfish] **************************************************************************************************************************************************************
fatal: [test]: FAILED! => {"changed": false, "details": "invalid literal for int() with base 8: 'u=rwx,go=rx'", "failed": true, "msg": "mode u=rwx,go=rx must be in octal form", "path": "payara41/", "state": "absent"}

Switching to Octal form helped. The problem occurs if your install failed (like you have no unzip or Java) and you do --retry

2) Added Init Script, which makes the work with it much easier (your vars are copied into a template)

3) Had a problem with running glassfish because of wrong permissions for domain1 folder - added a task to fix that using the set user

4) And in the end, I've added the run of GlassFish via InitScript, so after running this role - you got working GlassFish that you can use

Just tested it from my repo on Ubuntu 16.04,
Thanks!

dnikolayev committed:
Added Init Script for GlassFish

dnikolayev committed:
Changed permissions to Octal form because of bug on retry, added init script task copy, added fix for ownership domain1 directory, added initial run using init script

dpalomar commented:
Hi @dnikolayev , thanks for reporting.
I think that it is a weird issue that it works with octal mode only. I have tested with a heavy provisioned machine (ubuntu-16.04.1-server) and it works fine as you can see here [https://atlas.hashicorp.com/dpalomar/boxes/uc3mTiw](https://atlas.hashicorp.com/dpalomar/boxes/uc3mTiw).

The JDK1.8 and unzip are mandatory, because without java, the Glassfish/Payara doesn't run and unzip because really it is not a Glassfish server because is a fork named Payara with improvements and bugs resolved and tools that Oracle doesn't support, and the Payara site only offers the downloadable file as .zip.
So, if you use this role in a clean ubuntu 16 image, the unzip tool is not installed by default and the role fails to decompress the payara file.

dnikolayev commented:
Hi @dpalomar 

About unzip - I added it to my main playbook, so it was fine.
I was testing also on Ubuntu 16.04, but problems occur when you retry.

Also, I added init script, which simplifies work with Glassfish via ansible.
However, the provided recipe works very well for me now, and I have running Glassfish just by running this role in a playbook.

dpalomar commented:
Thanks @dnikolayev , I'd like maintain this role as standard as possible. Feel free to change whatever you need. 
I'll see if this init script is working for all escenarios or only yours before merge.
